src/overlays/actors/ovl_En_Death/z_en_death.c

```diff
@@ -50,230 +69,1683 @@ ActorInit En_Death_InitVars = {
+            this->actor.world.pos.y + yOffset, this->actor.world.pos.z, 0, yRot, 0, i);
+        if (this->miniDeaths[i] == NULL) {
+            Actor_Kill(&this->actor);
+        }
+        yRot += 0x3A00;
+        yOffset += 4.0f;
+    }
+
+    // Make the ith bat a child of the (i+1)th bat
+    for (i = 0; i < ARRAY_COUNT(this->miniDeaths) - 1; i++) {
+        this->miniDeaths[i]->actor.child = &this->miniDeaths[i + 1]->actor;
+    }
+
+    this->bodyMatAnim = Lib_SegmentedToVirtual(&gGomessBodyMatAnim);
+    this->coreMatAnim = Lib_SegmentedToVirtual(&gGomessCoreMatAnim);
+
+    if (CHECK_EVENTINF(EVENTINF_INTRO_CS_WATCHED_GOMESS)) {
+        // Watched intro cutscene
+        this->holdsScythe = true;
+        Actor_SetScale(&this->actor, 0.01f);
+        this->matAnimStep = 23;
+        this->actor.params = 20;
+        this->scytheScale = 1.0f;
+        EnDeath_SetupBeginWithoutCutscene(this);
+    } else {
+        // Start intro cutscene
+        this->inEarlyIntro = true;
+        this->actor.flags &= ~ACTOR_FLAG_TARGETABLE;
+        EnDeath_SetupPlayCutscene(this);
+    }
+}
+
+void EnDeath_Destroy(Actor* thisx, PlayState* play) {
+    EnDeath* this = THIS;
+
+    Collider_DestroySphere(play, &this->coreCollider);
+    Collider_DestroyCylinder(play, &this->bodyCollider);
+    Collider_DestroyQuad(play, &this->weaponCollider);
+    Collider_DestroyTris(play, &this->weaponSpinningCollider);
+    AudioSfx_StopByPos(&this->scytheScreenPos);
+    Effect_Destroy(play, this->effectIndex);
+}
+
+void EnDeath_DimLights(PlayState* play) {
+    EnvLightSettings* lightSettings;
+    s32 i;
+
+    play->envCtx.lightSettingOverride = 20;
+    play->envCtx.lightSetting = 20;
+    play->envCtx.prevLightSetting = 20;
+    play->envCtx.lightBlend = 1.0f;
+    lightSettings = &play->envCtx.lightSettingsList[20];
+
+    for (i = 0; i != 3; i++) {
+        lightSettings->light1Dir[i] = play->envCtx.lightSettings.light1Dir[i];
+        lightSettings->light2Dir[i] = play->envCtx.lightSettings.light2Dir[i];
+    }
+}
+
+void EnDeath_Float(EnDeath* this) {
+    Actor_PlaySfx_Flagged(&this->actor, NA_SE_EN_SHARP_FLOAT - SFX_FLAG);
+
+    if (this->floatTimer == 0) {
+        this->floatTimer = 40;
+    }
+    this->floatTimer--;
+    this->actor.world.pos.y = this->actor.home.pos.y + (1.0f - Math_CosS(this->floatTimer * 0x666)) * 7.5f;
+}
+
+s32 EnDeath_ProjectileApproaching(EnDeath* this, PlayState* play) {
+    Actor* projectileActor =
+        func_800BC270(play, &this->actor, 80.0f, 0x10000 | 0x2000 | 0x1000 | 0x800 | 0x80 | 0x20 | 0x10);
+    s16 ret;
+
+    if (projectileActor != NULL &&
+        (ret = Actor_WorldYawTowardActor(&this->actor, projectileActor) - this->actor.shape.rot.y,
+         ABS_ALT(ret) < 0x2000) &&
+        (ret = Actor_WorldPitchTowardPoint(projectileActor, &this->actor.focus.pos) - projectileActor->world.rot.x,
+         ABS_ALT(ret) < 0x3000)) {
+        return true;
```

Comment:
Is it possible to split up the conditional so the assignment doesn't happen inside?